doc/source/Python.rst

```diff
@@ -66,12 +66,65 @@ To install Pygrackle, you just need to invoke the following from the root direct
 
     ~/grackle $ pip install -v .
 
-You can configure the exact C and Fortran compilers that are used for this by manipulating the ``CC`` and ``FC`` environment variables.
-If you must pass extra compiler flags to all invocations of the C or Fortran compiler, you can use the ``CFLAGS`` or ``FFLAGS`` environment variable.
+Under this approach, pygrackle's python build backend, `scikit-build-core <https://scikit-build-core.readthedocs.io/en/latest/>`__, automatically builds the core Grackle library (:ref:`in a CMake build <cmake_build>`), packages the resulting library as part of the pygrackle package, and cleans up from the build.
+If a new enough version of CMake cannot be found, the above command will automatically download CMake (it will only be accessible during the build).
+
+While the above command should "just work," you have a few options for customizing the build:
+
+1. In most cases, you probably want to use the standard environment variables known to CMake:\ [#f1]_
+
+   - you can specifiy your choice of C and Fortran compilers that are used for this by manipulating the :envvar:`!CC` and :envvar:`!FC` environment variables.
+   - if you must pass extra compiler flags to all invocations of the C or Fortran compiler (this *shouldn't* really be necessary), you can use the :envvar:`!CFLAGS` or :envvar:`!FFLAGS` environment variable.
+   - to provide a hint about the location of the hdf5 library you can use :envvar:`!HDF5_ROOT` or :envvar:`!HDF5_DIR` (these behave similarly to the CMake variables of the same name).
+
+2. Alternatively, you can specify the values of CMake variables.
+   For the sake of argument, let's imagine that you want to assign the CMake variables ``CMAKE_C_COMPILER`` and ``HDF5_ROOT`` values of ``gcc-14`` and ``/path/to/hdf5``.
+   There are a few ways to do this:
+
+   .. tabs::
+
+      .. group-tab:: pip
+
+         If using a version of pip from before 23.1, you need to replace ``-C`` with ``--config-settings=`` in the following snippet.
+
+         .. code-block:: shell-session
+
+            ~/grackle $ pip install -v . -Ccmake.args=-DCMAKE_C_COMPILER=gcc-14;-DHDF5_ROOT=/path/to/hdf5
+
+      .. group-tab:: Environment
```

Comment:
The `Environment` tab is a bit confusing to me (both here and the one below). The `pip` tab ends with a code-block example for the user to execute, but this one doesn't.  I couldn't quite parse what one was supposed to do in the "Environment" option, but I think a code block example would help.